src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/synthetic_image_generator.py

```diff
@@ -1,62 +1,82 @@
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import glob
+import random
 from enum import Enum, auto
-from io import BytesIO
-from typing import Optional, cast
+from pathlib import Path
 
-import numpy as np
+from genai_perf import utils
 from PIL import Image
 
 
 class ImageFormat(Enum):
-    JPEG = auto()
     PNG = auto()
+    JPEG = auto()
 
 
 class SyntheticImageGenerator:
-    def __init__(
-        self,
+    """A simple synthetic image generator that generates multiple synthetic
+    images from the source images.
+    """
+
+    @classmethod
+    def create_synthetic_image(
+        cls,
         image_width_mean: int,
-        image_height_mean: int,
         image_width_stddev: int,
+        image_height_mean: int,
         image_height_stddev: int,
-        image_format: ImageFormat = ImageFormat.PNG,
-        rng: Optional[np.random.Generator] = None,
-    ):
-        self._image_width_mean = image_width_mean
-        self._image_height_mean = image_height_mean
-        self._image_width_stddev = image_width_stddev
-        self._image_height_stddev = image_height_stddev
-        self.image_format = image_format
-        self.rng = cast(np.random.Generator, rng or np.random.default_rng())
-
-    def __iter__(self):
-        return self
-
-    def _sample_random_positive_integer(self, mean: int, stddev: int) -> int:
+        image_format: ImageFormat,
+    ) -> str:
+        """Generate base64 encoded synthetic image using the source images."""
+        width = cls._sample_random_positive_integer(
+            image_width_mean, image_width_stddev
+        )
+        height = cls._sample_random_positive_integer(
+            image_height_mean, image_height_stddev
+        )
+
+        image = cls._sample_source_image()
+        image = image.resize(size=(width, height))
+
+        img_base64 = utils.encode_image(image, image_format.name)
+        return f"data:image/{image_format.name.lower()};base64,{img_base64}"
+
+    @classmethod
+    def _sample_source_image(cls):
```

Comment:
The synthetic image generator functionally chooses from only 4 images?
is that the correct?
is that enough randomness to enable good profiling?